itest/rust/src/object_tests/object_test.rs

```diff
@@ -948,6 +952,16 @@ pub mod object_test_gd {
         fn return_refcounted_as_object(&self) -> Gd<Object> {
             Gd::from_object(MockRefCountedRust { i: 42 }).upcast()
         }
+
+        #[func]
+        fn return_self() -> Gd<Self> {
+            Gd::from_object(Self)
+        }
+
+        #[func]
+        fn return_nested_self() -> Array<Gd<<Self as GodotClass>::Base>> {
+            Array::from(&[<Self as GodotClass>::Base::new_gd()])
```

Comment:
This could probably be:
```suggestion
            array![Self::return_self().upcast()]
```